### PR TITLE
Create configuration file

### PR DESCRIPTION
**Pull request by michaella23**
## Description

This PR adds a Configuration Markdown file to help users utilize this copilot-extension-template. Currently, there is text only. I wanted to know if you have a preference for file organization of screenshot images. Can I use the same images from the blog post? I have a few more questions, commented in the code on lines 12, 45, and 103.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] 📦 Chore
- [ ] ⏩ Revert

## Related Tickets & Documents

Relates to #16

## Mobile & Desktop Screenshots/Recordings

N/A

## Steps to QA

N/A

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📜 CONTRIBUTING.md
- [ ] 🙅 no documentation needed
- [x] ⚙️ CONFIGURATION.md

## [optional] Are there any post-deployment tasks we need to perform?

N/A

## [optional] What gif best describes this PR or how it makes you feel?

animated question mark with eyes rolling around

### CODE REVIEW

File: CONFIGURATION.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+## Create a GitHub app
+
+Head to your settings page on GitHub, and at the bottom, to the left, click on the **Developer Settings** link. This will bring you to your **GitHub apps**. (You can also directly navigate to your **GitHub apps** page at https://github.com/settings/apps.) Configure the following:
+
+### General Settings
+
+1. Enter a **GitHub App name**, e.g. my-copilot-extension
+2. Enter a URL for the homepage. This can be the same as the test URL for now.
+3. Set the **Callback URL** (currently required). This can be the same as the test URL for now. 
+<!-- ***** If these (Homepage URL and Callback URL) are indicated in the first part of the article, how best to reference it here? ***** -->
```
Comment (nickytonline) [inline CONFIGURATION.md:12]: You can use the screen shot from the blog post. In fact you can save all the images from the relevant sections.

File: CONFIGURATION.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+3. Select your extension and ask a question or perform an operation.
+
+4. The Copilot extension will return a response of "Welcome *your_github_username*! It looks like you asked the following question: *your_question*. This is a GitHub Copilot extension template, so it's up to you to decide what you want to implement to answer prompts."
+
+<!-- screenshot Copilot chat responding,  -->
+
+It won't respond to your specific question, as that functionality has not been implemented. This is where you can explore the preview SDK or integrate with a third-party service to provide more meaningful responses.
+
+<!-- ***** Is this to be included as well? *****
```
Comment (nickytonline) [inline CONFIGURATION.md:103]: So the only sections to omit from the blog post are the `Creating the Endpoint for Your Copilot Extension`, `A Real-World Example`, `Grab the GitHub Copilot Extension Template`, `Take a deep dive into Copilot extensions and the preview SDK` and the `Wrapping Up` sections of the blog post.

**Review by nickytonline — COMMENTED** [review]
Let me know if anything else isn't clear @michaella23. Once the images are in, feel free to take it out of draft mode and we can iterate from there. Thanks for working on this!

**Comment by nickytonline** [comment]
Also, currently in the README there is a getting started section.

CleanShot 2024-11-03 at 08 18 56

We could link to your file in the README with something like, To get up and running with your development environment, see the [Development Guide](./), or something like that.

```diff
- ## Getting Started
+ ## Installation
+
+ Run the following commands to install and start the application locally

npm install
npm run dev
open http://localhost:3000

- Use tunneling to expose your localhost to the internet via [VS Code's Port forwarding](https://code.visualstudio.com/docs/editor/port-forwarding), [ngrok](https://ngrok.com), or similar.

- You'll need to create a GitHub app for the Copilot extension to work and use the URL from the from the tunneling above in your GitHub app configuration.

- For a full breakdown of how to get started, check out Nick's Blog post, [Creating a GitHub Copilot Extension: A Step-by-Step Guide](https://nickyt.co/copilot-extension-guide).
+
+ ## Development Environment
+
+ To get up and running with your development environment, see the [Development Guide](./)
```

File: CONFIGURATION.md (hunk 1 above)
Comment (michaella23) [reply CONFIGURATION.md:12]: Can I make a directory for the screenshots? Something like `/docs/images`?

Comment (nickytonline) [reply CONFIGURATION.md:12]: Whatever you think makes sense. That sounds good to me. If we create a docs folder, consider putting the markdown file in there in the root as well.

**Comment by michaella23** [comment]
Hey @nickytonline! I've finished the next draft of the Development Guide. 

- [x] Created `CONFIGURATION.md`
- [x] Adapted text from article for this documentation
- [x] Added all relevant screenshots in a new directory - `./docs/images`
- [x] Moved `CONFIGURATION.md` to the root of the `docs` directory
- [x] Edited `README.md` to route the user to the Development Guide

Please let me know if there are any changes you'd like me to make!

File: docs/CONFIGURATION.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+```
+const port = 3000;
+console.log(`Server is running on port ${port}`);
+
+serve({
+  fetch: app.fetch,
+  port,
+});
+```
```
Comment (nickytonline) [inline docs/CONFIGURATION.md:25]: This is how I wrote the blog post, but I'm going to update it to this.

```suggestion
1. If using Visual Studio Code (VS Code), enable [port forwarding](https://code.visualstudio.com/docs/editor/port-forwarding). Note that the port is private by default - a good thing - but for this use case you need to set it to public.

![Port forwarding tab in the bottom dock of VS Code](./images/port_forwarding.png)

Alternatively, use tools like [cloudflared](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/create-local-tunnel/) or [ngrok](https://ngrok.com/) to expose a public URL.

The application is set up to run on port 3000, so we'll forward port 3000.
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
One small change request to the docs, and also, let's renamed CONFIGURATION.md to SETUP.md or DEVELOPMENT_SETUP.md. I'm good with either.

Thanks again for working on this!

**Comment by michaella23** [comment]
> One small change request to the docs, and also, let's renamed CONFIGURATION.md to SETUP.md or DEVELOPMENT_SETUP.md.

Hey @nickytonline, I've made suggested changes. 🙂

File: README.md [hunk 4]
```diff
@@ -2,16 +2,16 @@
...
 open http://localhost:3000
 ```
 
-Use tunneling to expose your localhost to the internet via [VS Code's Port forwarding](https://code.visualstudio.com/docs/editor/port-forwarding), [ngrok](https://ngrok.com), or similar.
-
-You'll need to create a GitHub app for the Copilot extension to work and use the URL from the from the tunneling above in your GitHub app configuration.
+## Development Environment
 
-For a full breakdown of how to get started, check out Nick's Blog post, [Creating a GitHub Copilot Extension: A Step-by-Step Guide](https://nickyt.co/copilot-extension-guide).
+To get up and running with your development environment, see the [Development Guide](./docs/CONFIGURATION.md).
```
Comment (nickytonline) [inline README.md:17]:
```suggestion
To get up and running with your development environment, see the [Development Guide](./docs/DEVELOPMENT_SETUP.md).
```

**Review by nickytonline — APPROVED** [review]
Thanks for this @michaella23! :shipit:

**Pull request merged**